### PR TITLE
Fix permadiff on google_vpc_access_connector resource with breaking changes (min_instances, max_instances, min_throughput, max_throughput fields)

### PR DESCRIPTION
**Pull request by SarahFrench**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/15278

When reviewing this PR please see:
- This comment for context about this PR : https://github.com/GoogleCloudPlatform/magic-modules/pull/10313#issuecomment-2159065620
- This comment about the decision to make the two fields conflict in the schema: https://github.com/GoogleCloudPlatform/magic-modules/pull/10313#issuecomment-2159101760
- Note to the reviewer when I marked this ready for review: https://github.com/GoogleCloudPlatform/magic-modules/pull/10313#issuecomment-2161335142

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
vpcaccess: removed default values for `min_throughput` and `min_instances` fields on `google_vpc_access_connector` and made them default to values returned from the API when not provided by users
```

```release-note:breaking-change
vpcaccess: added a conflicting fields restriction between `min_throughput` and `min_instances` fields on `google_vpc_access_connector`
```

```release-note:breaking-change
vpcaccess: added a conflicting fields restriction between `max_throughput` and `max_instances` fields on `google_vpc_access_connector`
```

### CODE REVIEW

**Comment by SarahFrench** [comment]
As a recap of this PR's stages:

1. Originally contained only an acceptance test demonstrating the permadiff in the linked issue [f7bce51](https://github.com/GoogleCloudPlatform/magic-modules/pull/10313/commits/f7bce5173bc377313e518d2a693d6c6987924856) - [test failure due to permadiff here](https://github.com/GoogleCloudPlatform/magic-modules/pull/10313#issuecomment-2025212589)

2. Updated the provider (@ [26ca216](https://github.com/GoogleCloudPlatform/magic-modules/pull/10313/commits/26ca216d872fe90bc2bad9dc607a68609823f1a3)) to:
- remove default values on min_ and max_throughput, then needed to make those fields default from the API because all 4 fields are always present
- update existing tests to no longer rely on the default values for min_ and max_throughput
- update the tests  (in https://github.com/GoogleCloudPlatform/magic-modules/pull/10313/commits/fe60725f36152a6e2c469bbc64c6286be0abddee )related to the these fields' weird behaviour/the permadiff to show that [setting all 4 fields results in a permadiff](https://github.com/GoogleCloudPlatform/magic-modules/blob/fe60725f36152a6e2c469bbc64c6286be0abddee/mmv1/third_party/terraform/services/vpcaccess/resource_vpc_access_connector_test.go#L34) and that [when 2 fields are set the other 2 fields are present](https://github.com/GoogleCloudPlatform/magic-modules/blob/fe60725f36152a6e2c469bbc64c6286be0abddee/mmv1/third_party/terraform/services/vpcaccess/resource_vpc_access_connector_test.go#L57).

**Comment by SarahFrench** [comment]
In the commits above I stop users from being able to include both min_throughput and min_instance (same for max_) by making fields conflict. Previously users would experience a permadiff, but this change means that users will fix their config to avoid the permadiff.

**Comment by SarahFrench** [comment]
Ah, we cannot have acc tests that don't reach out to an API run in VCR:

```
    test_utils.go:108: no cassette found on disk for TestAccVPCAccessConnector_vpcAccessConnectorThroughput_combiningThroughputAndInstancesFields_conflict, please replay this testcase in recording mode - open /workspace/.ci/magician/cassettes/beta/TestAccVPCAccessConnector_vpcAccessConnectorThroughput_combiningThroughputAndInstancesFields_conflict.seed: no such file or directory

```

**Comment by SarahFrench** [comment]
👋 Hi @roaks3 - I requested you as a reviewer because this work's [based on a PR you reviewed ages ago](https://github.com/GoogleCloudPlatform/magic-modules/pull/10130#issuecomment-1991771276). This PR's changes are something I plan to be in 6.0.0. Although I avoid breaking changes due to making the fields O+C when removing their defaults, I believe adding the conflicting fields restriction to the schema makes it a breaking change.

Currently [the test here](https://github.com/GoogleCloudPlatform/magic-modules/pull/10313#issuecomment-2161216440) is failing due a CIDR issue. Do you know a quick fix to address this?

**Review by roaks3 — APPROVED** [review]
LGTM, sorry for the delay.

* Agreed that the `conflicts` configs that were added would probably be considered breaking
* I'm not sure about the test. It uses the default network, which can get crowded wrt IP space, but it seems like there should not be overlap with other tests. There was a leftover access connector in the project that I deleted, so might be worth trying one more time.

**Comment by SarahFrench** [comment]
Thanks, I'll re-run the tests : /gcbrun

**Comment by SarahFrench** [comment]
Hopefully this change (https://github.com/GoogleCloudPlatform/magic-modules/pull/10995) will address the test failure, and I can then rebase this PR to include those changes

**Comment by SarahFrench** [comment]
👆 going to merge to 6.0 feature branch

**Comment by SarahFrench** [comment]
I think I need to add an upgrade guide entry, but I believe that's something that goes into main instead of the feature branch. I'll prepare another PR soon.

**Comment by SarahFrench** [comment]
I realised that upgrade guides are being merged to the feature branch instead of main, so re-requested review so a second pair of eyes can check the upgrade guide changes. Thanks!

File: mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown
Comment (roaks3) [inline mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown]: Nit: Extra period

File: mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown
Comment (roaks3) [inline mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown:134]: Nit: "added"

Also, should the ending be "and `max_throughput` and `max_instances`"?

File: mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown
Comment (roaks3) [inline mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown]: Same here, I think it should be "and one of `max_throughput` and `max_instances`"

**Review by roaks3 — COMMENTED** [review]
I'm not 100% sure, but I think the guide entry describes conflicts between the wrong fields, which we'd want to fix

File: mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown
Comment (SarahFrench) [reply mmv1/third_party/terraform/website/docs/guides/version_6_upgrade.html.markdown:134]: Yes, thank you! 🤦  I was a little too copy/paste happy here

**Comment by SarahFrench** [comment]
Thanks Ryan! Given that the old approval is still in place after adding changes to the upgrade guide, I'll go ahead and merge this PR once the automation finishes

**Comment by SarahFrench** [comment]
That test failure is unrelated to the PR, and [has been failing in the nightlies for a while](https://hashicorp.teamcity.com/test/7385344304796565941?currentProjectId=TerraformProviders_GoogleCloud_GOOGLE_BETA_NIGHTLYTESTS&expandTestHistoryChartSection=true&expandedTest=build%3A%28id%3A189091%29%2Cid%3A2000000013).

Merging!

**Pull request merged**
